modules/nixos/impermenance.nix

```diff
@@ -0,0 +1,61 @@
+{ config
+, lib
+, ...
+}:
+let
+  inherit (lib) mkIf mkEnableOption;
+  cfg = config.custom.impermanence;
+in
+{
+  options.custom.impermanence = {
+    enable = mkEnableOption "impermenance";
+  };
+
+  config = mkIf cfg.enable {
+    boot.initrd.systemd.enable = true;
+    boot.initrd.systemd.services.reset-root = {
+      description = "Backup & reset root subvolume";
+      wantedBy = [
+        "initrd.target"
+      ];
+      after = [
+        "systemd-cryptsetup@enc.service" # luks & tpm
+      ];
+      before = [
+        "sysroot.mount"
+      ];
+      unitConfig.DefaultDependencies = "no";
+      serviceConfig.Type = "oneshot";
+      script = ''
+        # Mount the LUKS volume "crypted"
+        mkdir -p /btrfs_tmp
+        mount -o subvol=/ /dev/mapper/crypted /btrfs_tmp
```

Comment:
Can probably use `fileSystems."/".device` instead of hard-coding `/dev/mapper/crypted`

Reply:
Might also be able to extract the subvol name instead of assuming it is `root`?